### ISSUE TITLE
Bezier Pointer-Snap to Top

### ISSUE THREAD

**Issue by kylamon1** [thestonefox/VRTK#640]
We need an option for the bezier pointer to "snap" to the top of an object if we are pointing at/near the top of the object.

Example: in Test Scene 038, Dash Teleport, it is quite cumbersome to climb the towers because the bezier pointer wants to constantly fall down between each of the towers. It would be great if the bezier pointer instead of snapping to the ground at that spot if it was withing say 1 unit(maybe make this an adjustable variable) from the top of the object then it would put the cursor at the top of the tower instead of the bottom of the tower.

EDIT:
This could also be implemented into the simple pointer as well.

EDIT number 2:
![pointer_snap to top](https://cloud.githubusercontent.com/assets/2676704/19248578/f2c8bf28-8efe-11e6-83ce-38c9568b7335.jpg)

Proposed method:
When your raycast forward hits an object, if the option is turned on:
1. do Raycast UP an adjustable amount, "var SnapHeight" (say 1 unit)
2. From that position do another raycast Forward by an adjustable amount, "var playerDiameter". If this raycast hits another object, then the snap-to-top failed as the space is not big enough. If it does not hit an object then the snap-to-top succeeded.
3.  To find the "final Position" send raycast down from the midpoint of raycast from number 2. This midpoint would be the radius of the player...aka in the middle of the available space. 

Possible Problems:
1. I dont know if this would have issues with the headset being over the "air" and dropping them to the ground...see PlayerPresence(PlayerBodyPhysics).
2. Would we need another raycast(or 2) to check if the player would fit on the ledge or if they are too tall. This is probably beyond the scope of this particular issue.

**Comment by kylamon1** [comment]
Actually I dont think you would need that raycast up.  Just use the jointPosition from the projectBeamForward, then add the variable snapHeight to the y coordinate of the joint position.

Im trying to dig through the code, but my coding skills are not up to the caliber of you guys.

**Comment by thestonefox** [comment]
This sounds like a separate pointer concept rather than trying to bolt it on to the existing pointers.

**Comment by kylamon1** [comment]
Hey stonefox I dont have any experience with gitHub and the fork/pull stuff.  I was able to put togeather was I was planning, but I'm not sure how to share it with you. I'm sure the code is not without its bugs, but it does work as I imagined it would. 

https://drive.google.com/file/d/0B_bIAFFBaaheUzlma2FibnlZeFk/view?usp=sharing

Give this a try.  This would be a modification of the Bezier Pointer script.  I made my own class so I didnt overwrite the VRTK one I had in Unity.
